### ISSUE TITLE
ADD COLUMN fails in object array

### ISSUE THREAD

**Issue by burnes** [crate/crate#4360]
Don't know if this is a bug or an unsupported feature. I didn't find any further description in the reference documentation.

Crate 0.56.0

```sql
CREATE TABLE t (
    id STRING PRIMARY KEY,
    attributes ARRAY(
        OBJECT(STRICT) AS (
            name STRING
        )
    )
);
```

```sql
ALTER TABLE t ADD column attributes['test'] BOOLEAN;
```

Stack Trace:

```
SQLActionException: BAD_REQUEST 4000 MapperParsingException: Mapping definition for [attributes] has unsupported parameters: [inner : {dynamic=true, properties={description={type=string, index=not_analyzed}, is_available_for_filtering={type=boolean}, label={type=string, index=not_analyzed}, name={type=string, index=not_analyzed}, settings={dynamic=false, properties={isfilter={type=boolean}, mandatory={type=boolean}, show={type=boolean}}}, type={type=string, index=not_analyzed}, validation={dynamic=strict, properties={type_relation={dynamic=strict, properties={schemas={type=array, inner={type=string, index=not_analyzed, doc_values=false}}}}}}}}] at org.elasticsearch.index.mapper.DocumentMapperParser.checkNoRemainingFields(DocumentMapperParser.java:181) at org.elasticsearch.index.mapper.DocumentMapperParser.checkNoRemainingFields(DocumentMapperParser.java:175) at org.elasticsearch.index.mapper.object.ObjectMapper$TypeParser.parseProperties(ObjectMapper.java:307) at org.elasticsearch.index.mapper.object.ObjectMapper$TypeParser.parseObjectOrDocumentTypeProperties(ObjectMapper.java:218) at org.elasticsearch.index.mapper.object.RootObjectMapper$TypeParser.parse(RootObjectMapper.java:139) at org.elasticsearch.index.mapper.DocumentMapperParser.parse(DocumentMapperParser.java:128) at org.elasticsearch.index.mapper.DocumentMapperParser.parse(DocumentMapperParser.java:109) at org.elasticsearch.index.mapper.MapperService.parse(MapperService.java:514) at org.elasticsearch.cluster.metadata.MetaDataMappingService$PutMappingExecutor.applyRequest(MetaDataMappingService.java:257) at org.elasticsearch.cluster.metadata.MetaDataMappingService$PutMappingExecutor.execute(MetaDataMappingService.java:230) at org.elasticsearch.cluster.service.InternalClusterService.runTasksForExecutor(InternalClusterService.java:468) at org.elasticsearch.cluster.service.InternalClusterService$UpdateTask.run(InternalClusterService.java:772) at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:231) at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:194) at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) at java.lang.Thread.run(Thread.java:745)
```

**Comment by mikethebeer** [comment]
@burnes this is a bug. We are going to look at this in detail and report back.

**Comment by bogensberger** [comment]
@burnes thanks for reporting it. Fixed it with https://github.com/crate/crate/pull/4406. Will also be released with the next 0.56 release

**Comment by mfussenegger** [comment]
Closing this, the fix was released with 0.56.4
